getsmarter_api_clients/geag.py

```diff
@@ -23,8 +25,107 @@ def get_terms_and_policies(self):
+        self,
+        payment_reference,
+        address_line1,
+        city,
+        postal_code,
+        state,
+        state_code,
+        country,
+        country_code,
+        first_name,
+        last_name,
+        email,
+        date_of_birth,
+        terms_accepted_at,
+        currency,
+        order_items,
+        address_line2=None,
+        mobile_phone=None,
+        work_experience=None,
+        education_highest_level=None
+    ):
         """
         Create an allocation (enrollment) through GEAG.
+
+        Args:
+            payment_reference (str): Reference used by enterprise partner
+            when payment is made to GetSmarter
+            address_line1 (str): Address Line 1
+            city (str): City
+            postal_code (str): Postal code
+            state: State
+            state_code: State code
+            country (str): Country
+            country_code (str): Country code,
+            first_name (str): First name
+            last_name (str): Last name
+            email (str): Email
+            date_of_birth (str): Date of birth
+            terms_accepted_at (str): ISO 8601 timestamp of
+            when the terms and policies were accepted
+            currency: One of ['USD', 'GBP', 'ZAR', 'EUR', 'AED',
+            'SGD', 'HKD', 'SAR', 'INR', 'CAD']
+            order_items: Items ordered
+            address_line2: Adress Line 2
+            mobile_phone: Mobile phone number
+            work_experience: One of ['None', '1 to 5 years',
+            '5 to 15 years', 'More than 15 years']
+            education_highest_level: Highest education reached by the learner
+
+            **Example payload**
+                {
+                    'paymentReference': 'GS-12304',
+                    'addressLine1': 'Oak Glen',
+                    'city': 'Cape Town',
+                    'postalCode': '7570',
+                    'country': 'South Africa',
+                    'countryCode': 'ZA',
+                    'firstName': 'Jan',
+                    'lastName': 'Pan',
+                    'email': 'janpan@gs.com',
+                    'dateOfBirth': '2021-05-12',
+                    'termsAcceptedAt': '2021-05-21T17:32:28Z',
+                    'currency': 'ZAR',
+                    'orderItems': [
+                        {
+                            'productId': 'product_id',
+                            'quantity': 1,
+                            'normalPrice': 1000,
+                            'discount': 500,
+                            'finalPrice': 500
+                        }
+                    ],
+                }
         """
-        return NotImplementedError()
+        url = f'{self.api_url}/allocations'
+
+        payload = {
+            'paymentReference': payment_reference,
+            'addressLine1': address_line1,
+            'city': city,
+            'postalCode': postal_code,
+            'state': state,
+            'stateCode': state_code,
```

Comment:
nit/question: These two fields (state and stateCode) are technically optional per the swagger docs. Should they be moved down to "optional fields" and fallback to `None` in the method kwargs similar to the other optional fields?

Though it is odd they require other address-related fields other than state (e.g., country).

Reply:
<img width="717" alt="image" src="https://user-images.githubusercontent.com/2828721/182663366-5cf7f178-fab8-46eb-8fb3-05fc190e4e70.png">

Reply:
The API returns a 400 without these two fields, so the docs might not match reality :(.

Reply:
Welp. Got it, thanks!